src/snowflake/snowpark/session.py

```diff
@@ -631,6 +645,21 @@ def cte_optimization_enabled(self, value: bool) -> None:
             )
         self._cte_optimization_enabled = value
 
+    @eliminate_numeric_sql_value_cast_enabled.setter
+    @experimental_parameter(version="1.20.0")
+    def eliminate_numeric_sql_value_cast_enabled(self, value: bool) -> None:
+        """Set the value for eliminate_numeric_sql_value_cast_enabled"""
+
+        if value in [True, False]:
```

Comment:
You can only send telemetry when it is false for now

Reply:
i actually do want to send telemetry for both values whenever user configures it, the telemetry method I added for this parameter is able to deal with both True and False.